wire-library/wire-runtime/src/jvmMain/kotlin/com/squareup/wire/internal/JsonIntegration.kt

```diff
@@ -171,6 +173,25 @@ abstract class JsonIntegration<F, A> {
     }
   }
 
+  /** Encodes a unsigned value without quotes, like `123`. */
+  private object UnsignedIntAsNumberJsonFormatter : JsonFormatter<Int> {
+    // 2^32, used to convert sint32 values >= 2^31 to unsigned decimal form
+    private const val power32 = 1L shl 32
+
+    override fun fromString(value: String): Int {
+      val doubleValue = value.toDouble()
+      return if (doubleValue < 0.0) doubleValue.toInt()
+      else doubleValue.toUInt().toInt()
```

Comment:
Do we use `UInt` elsewhere?

Reply:
It's not ABI stable so we shouldn't use it regardless

Reply:
fixed